vm/message/src/lib.rs

```diff
@@ -43,3 +43,50 @@ pub trait Message {
     /// sets the gas premium.
     fn set_gas_premium(&mut self, prem: TokenAmount);
 }
+
+impl Message for ShimMessage {
+    fn from(&self) -> Address {
+        Address::from(self.from)
+    }
+    fn to(&self) -> Address {
+        Address::from(self.to)
+    }
+    fn sequence(&self) -> u64 {
+        self.sequence
+    }
+    fn value(&self) -> TokenAmount {
+        TokenAmount::from(&self.value)
+    }
+    fn method_num(&self) -> MethodNum {
+        self.method_num
+    }
+    fn params(&self) -> &RawBytes {
+        &self.params
+    }
+    fn gas_limit(&self) -> i64 {
+        self.gas_limit as i64
+    }
+    fn set_gas_limit(&mut self, token_amount: i64) {
```

Comment:
Could it be unsigned?

Reply:
Yeah. Patching this is a bit of work, though. I'll create a new tracking issue.

Reply:
Tracking issue: https://github.com/ChainSafe/forest/issues/2574